File: app/Models/EmploiDuTemps.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class EmploiDuTemps extends Model
{
    use HasFactory;

    protected $table = 'emplois_du_temps';

    protected $fillable = [
        'classe_id',
        'matiere_id',
        'professeur_id',
        'jour',
        'heure_debut',
        'heure_fin',
        'salle'
    ];

    public function classe()
    {
        return $this->belongsTo(Classe::class);
    }

    public function matiere()
    {
        return $this->belongsTo(Matiere::class);
    }

    public function professeur()
    {
        return $this->belongsTo(Professeur::class);
    }
}

File: parent_frontend/lib/models/emploi_du_temps.dart
class EmploiDuTemps {
  final int id;
  final int classeId;
  final int matiereId;
  final int professeurId;
  final String jour;
  final String heureDebut;
  final String heureFin;
  final String? salle;
  final Map<String, dynamic>? classe;
  final Map<String, dynamic>? matiere;
  final Map<String, dynamic>? professeur;

  EmploiDuTemps({
    required this.id,
    required this.classeId,
    required this.matiereId,
    required this.professeurId,
    required this.jour,
    required this.heureDebut,
    required this.heureFin,
    this.salle,
    this.classe,
    this.matiere,
    this.professeur,
  });

  factory EmploiDuTemps.fromJson(Map<String, dynamic> json) {
    return EmploiDuTemps(
      id: json['id'],
      classeId: json['classe_id'],
      matiereId: json['matiere_id'],
      professeurId: json['professeur_id'],
      jour: json['jour'],
      heureDebut: json['heure_debut'],
      heureFin: json['heure_fin'],
      salle: json['salle'],
      classe: json['classe'],
      matiere: json['matiere'],
      professeur: json['professeur'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'classe_id': classeId,
      'matiere_id': matiereId,
      'professeur_id': professeurId,
      'jour': jour,
      'heure_debut': heureDebut,
      'heure_fin': heureFin,
      'salle': salle,
      'classe': classe,
      'matiere': matiere,
      'professeur': professeur,
    };
  }
}
